ha/alert/node_alert_monitor.py

```diff
@@ -27,9 +30,53 @@ def __init__(self):
         Init node alert monitor
         """
         super(NodeAlertMonitor, self).__init__()
+        self.process = SimpleCommand()
+
+    def _get_online_nodes(self):
+        """
+        Get list of online nodes ids.
+        """
+        online_nodes_xml = self.process.run_cmd("crm_mon --as-xml")
+        # saving the xml file
+        with open('nodes.xml', 'w+') as f:
+            f.write(online_nodes_xml[0])
```

Comment:
Can't we get the work done without writing to a file?

Reply:
No we can't, for ET.parse('nodes.xml') method it needs file hence written in that way.

Reply:
@shindeam Please look at the code ha/core/controllers/pcs/cluster_status.py. You don't need to save output to a file.

Reply:
Git it, taken code and updated our code by referencing it.